### PR TITLE
CORDA-2775: Simplify SourceClassLoader logic for DJVM.

### PR DESCRIPTION
**Pull request by chrisr3**
Simplify the `SourceClassLoader` logic to make it easier for people to use DJVM as a library.

Removing `SandboxSourceClassLoader` means that we don't need `AbstractSourceClassLoader` any more either, leaving us with just `SourceClassLoader` again.

### CODE REVIEW

File: djvm/djvm/src/test/java/net/corda/djvm/execution/SandboxExecutorJavaTest.java [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
     @Test
     public void testTransaction() {
+        //TODO: Transaction should not be a pinned class! It needs
```
Comment (tlil) [inline djvm/djvm/src/test/java/net/corda/djvm/execution/SandboxExecutorJavaTest.java]: This will be addressed in a separate PR I take it?

Comment (chrisr3) [reply djvm/djvm/src/test/java/net/corda/djvm/execution/SandboxExecutorJavaTest.java]: Yes, it's _far_ too large a body of work to address now.
